src/NUnitFramework/framework/Attributes/ParallelizableAttribute.cs

```diff
@@ -51,6 +52,30 @@ public ParallelizableAttribute(ParallelScope scope) : base()
             Properties.Add(PropertyNames.ParallelScope, scope);
         }
 
+        /// <summary>
+        /// Overridden to check for invalid combinations of settings
+        /// </summary>
+        /// <param name="test"></param>
+        public override void ApplyToTest(Test test)
+        {
+            base.ApplyToTest(test);
+
+            if (test.RunState == RunState.NotRunnable)
+                return;
+
+            if (_scope.HasFlag(ParallelScope.Self) && _scope.HasFlag(ParallelScope.None))
+                MarkTestInvalid(test, "Test may not be both Parallel and NonParallel");
```

Comment:
Good catch. They were in an earlier version of the enum. I think I prefer "Test may not be both parallel and non-parallel".